### PR TITLE
DM-34751: Integrate felis into lsst build infrastructure

### PR DESCRIPTION
**Pull request by andy-slac**
`felis` now builds with scons and is configured it for `eups`. Python code moved to `python/felis`. Bunch of unit tests defined for main modules. Run code through `black` and `isort`. Added type annotations. Couple of bugs fixed.

### CODE REVIEW

**Review by timj — APPROVED** [review]
All looks great.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+name: build_and_test
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply .github/workflows/build.yaml:1]: 👍 I'll add `.yamllint.yaml`.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      matrix:
+        python-version: ["3.9", "3.10"]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Need to clone everything for the git tags.
+          fetch-depth: 0
+
+      - uses: conda-incubator/setup-miniconda@v2
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply .github/workflows/build.yaml:24]: Probably not, this is just copy and paste, I need to find an example of using Python without conda.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+[build-system]
+requires = ["setuptools", "lsst-versions"]
+build-backend = "setuptools.build_meta"
+
+[tool.towncrier]
+    package = "felis"
+    package_dir = "python"
+    filename = "doc/felis/CHANGES.rst"
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply pyproject.toml:8]: I think docs will need a separate ticket, I'll leave it as it is for now, but will make a ticket for updating it. (and change doc to docs here for now).

File: python/felis/cli.py [hunk 4]
```diff
@@ -14,29 +16,33 @@
...
 from sqlalchemy import create_engine
 
+from . import DEFAULT_CONTEXT, DEFAULT_FRAME, __version__
 from .model import Visitor, VisitorBase
-from .tap import TapLoadingVisitor, Tap11Base, init_tables
+from .tap import Tap11Base, TapLoadingVisitor, init_tables
 from .utils import ReorderingVisitor
-from . import __version__, DEFAULT_CONTEXT, DEFAULT_FRAME
 
 logging.basicConfig(level=logging.INFO)
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply python/felis/cli.py:39]: I moved `basicConfig` inside `cli()`. The level is left at INFO for now, I'll change default later when `--log-level` is added.

File: python/felis/db/sqltypes.py [hunk 5]
```diff
@@ -15,12 +17,15 @@
 # GNU General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License
-# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from sqlalchemy.ext.compiler import compiles
-from sqlalchemy import SmallInteger, Float
-from sqlalchemy import types
+import builtins
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply python/felis/db/sqltypes.py:22]: There is  a `def int()` in the code below and that confuses mypy when it sees `length: int` in other places.

File: .github/workflows/build.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (andy-slac) [reply .github/workflows/build.yaml]: Found an example in `pex_config`.

**Pull request merged**